scanner/src/main/kotlin/utils/FileListResolver.kt

```diff
@@ -47,23 +44,16 @@ internal class FileListResolver(
         }
     }
 
-    fun has(provenance: KnownProvenance): Boolean = storage.hasFile(provenance)
+    fun has(provenance: KnownProvenance): Boolean = storage.hasData(provenance)
 }
 
 private fun ProvenanceFileStorage.putFileList(provenance: KnownProvenance, fileList: FileList) {
-    val tempFile = createOrtTempFile(prefix = "file-list", suffix = ".yml")
-
-    tempFile.writeValue(fileList)
-    putFile(provenance, tempFile)
-
-    tempFile.delete()
+    putData(provenance, fileList.toYaml().byteInputStream())
```

Comment:
Would be nice if we could avoid serializing the list to memory first and instead directly serialize to the stream. But IIRC Jackson can only serialize to an output stream and converting that to an input stream requires to use the `PipedInputStream` where I had problems the last time I tried to use it.